### PR TITLE
Backend graphql setup

### PR DESCRIPTION
**Pull request by danielperev**
**Add backend GraphQL setup**:
   1. GraphQL schema setup in `main\resources\graphql\schema.graphqls` 
   2. Folder structure: controllers in `main\controllers\` domain/models/definitions in `main\domain\`
   3. Can use localhost:9000/graphql (POST) for schema fetching (GraphQL introspection enabled in application.yml) and querying
   4. GraphQLConfig.java enables logging of errors for debugging purposes
Added example requests for user/prompt fetching in `main\resources\graphql\insomnia_requests_example.json`
Currently just returning raw data I made.

**Add testing with maven-surefire**:
This is apparently the official way to test with maven. Just run `mvn test` in backend folder.

### CODE REVIEW

File: backend/src/main/resources/application.yml [hunk 1]
```diff
@@ -3,6 +3,16 @@ spring:
         name: jobwait
     datasource:
         driver-class-name: org.postgresql.Driver
-        url: jdbc:postgresql://localhost:5432/mydatabase
+        url: jdbc:postgresql://localhost:5432/test
```
Comment (pecasdev) [inline backend/src/main/resources/application.yml]: does this work? i think the name has to be the same as the name in the [compose.yaml](https://github.com/pecasdev/jobwait/blob/882725dca98f46dc809d48cdfe53b3ccb6ad6217/backend/compose.yaml#L8)

File: backend/src/main/resources/graphql/schema.graphqls [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    response: Response
+}
+
+type IntResponse {
+    intResponse: Int
+}
+
+type StrResponse {
+    strResponse: String
+}
```
Comment (pecasdev) [inline backend/src/main/resources/graphql/schema.graphqls]: make intResponse and strResponse required

File: backend/src/main/resources/application.yml (hunk 1 above)
Comment (danielperev) [reply backend/src/main/resources/application.yml]: sorry ill remove this was from really early stuff when i was trying to get flyway to work.

File: backend/src/main/resources/graphql/schema.graphqls (hunk 2 above)
Comment (danielperev) [reply backend/src/main/resources/graphql/schema.graphqls]: good point

**Review by pecasdev — APPROVED** [review]

**Pull request merged**